dbms/src/IO/IReadableWriteBuffer.h

```diff
@@ -5,13 +5,14 @@
 namespace DB
 {
 
+
 struct IReadableWriteBuffer
 {
     /// At the first time returns getReadBufferImpl(). Next calls return nullptr.
     inline std::shared_ptr<ReadBuffer> tryGetReadBuffer()
     {
-        if (!can_reread)
```

Comment:
@alexey-milovidov  take a look here
The compiler complains if possible null dereference, do you have any idea why it is not satisfied?

Reply:
At what exact location possible dereference happens?

Reply:
Will post here the exact warning

Reply:
@alexey-milovidov 
Here you go
In file included from /home/user/Development/ClickHouse/dbms/src/IO/MemoryReadWriteBuffer.h:5,
                 from /home/user/Development/ClickHouse/dbms/src/IO/tests/gtest_cascade_and_memory_write_buffer.cpp:9:
/home/user/Development/ClickHouse/dbms/src/IO/IReadableWriteBuffer.h: In static member function ‘static _Res std::_Function_handler<_Res(_ArgTypes ...), _Functor>::_M_invoke(const std::_Any_data&, _ArgTypes&& ...) [with _Res = std::shared_ptr<DB::WriteBuffer>; _Functor = checkHTTPHandlerCase(size_t, size_t)::<lambda(const WriteBufferPtr&)>; _ArgTypes = {const std::shared_ptr<DB::WriteBuffer>&}]’:
/home/user/Development/ClickHouse/dbms/src/IO/IReadableWriteBuffer.h:14:14: error: potential null pointer dereference [-Werror=null-dereference]
   14 |         if (!can_reread)
      |              ^~~~~~~~~~
/home/user/Development/ClickHouse/dbms/src/IO/IReadableWriteBuffer.h:14:14: error: potential null pointer dereference [-Werror=null-dereference]
At global scope: